src/main/java/frc/robot/UserInterface.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Team 4909, Bionics
+ * Billerica Memorial High School
+ *
+ * Copyright:
+ *   2021 Bionics
+ *
+ * License:
+ *   MIT: https://opensource.org/licenses/MIT
+ *   See the LICENSE file in the project's top-level directory for details.
+ */
+
+package frc.robot;
+
+import java.lang.Object;
+import java.util.HashMap;
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj2.command.button.JoystickButton;
+import frc.peyton.Drivetrain;
+import frc.bionic.swerve.command.DriveWithJoystick;
+
+public class UserInterface
+{
+  // Registry of objects that our user interface can operate on
+  private static HashMap<String, Object> objectRegistry = new HashMap<String, Object>();
```

Comment:
I wonder if we could find a more direct concise way of passing in needed objects. Usually casting is code smell.

Reply:
If casting is used inappropriately, it can be ugly. Casting exists for a reason, though, in strictly-typed languages. Collections are really useful, but are limited in Java. A collection is forced to be of all the same type objects... but all objects descend from java.lang.Object. By having a collection (HashMap, in this case) that contains entries of type Object, various specific object types can be stored. The casts do not alter the object; only allow putting what is known to be a specific type object back into a variable of that type.

Reply:
> I wonder if we could find a more direct concise way of passing in needed objects. Usually casting is code smell.

Haven't been keeping tabs on the recent robot code, but is there any intermediate superclass you could use to then have a small number of registries to avoid the casting while allowing dynamic lookups? Like I think you could easily have an interface implemented by all subsystems that allow for analog joystick input and then allow avoid the casting concerns aforementioned. Granted, it's a few more registries but I've found that pattern to find a balance between both options in some of my projects.